test/conformance/suite_test.go

```diff
@@ -54,6 +54,14 @@ var (
 // -----------------------------------------------------------------------------
 
 func TestMain(m *testing.M) {
+	var code int
+	defer func() {
+		if r := recover(); r != nil {
+			fmt.Printf("ERROR: %v\n", r)
+			code = 1
+		}
```

Comment:
Would `recover` produce a more readable output? I'm wondering if we won't loose information when printing this way.

Reply:
The rationale for this change was an issue reported by `gocritic` - that due to `os.Exit` deffers won't execute. So let's ensure that `os.Exit` is executed after all defers (as a last defer). We lose stack trace from `panic` called in `exitOnErr` it's not very important I think

Reply:
We can always `defer` only the `os.Exit()` as the last one. IMHO panic stack traces are very helpful in debugging when something goes wrong.

Reply:
> We can always defer only the os.Exit() as the last one. 

Unfortunately, it doesn't, it will exit silently.

But stack trace can be ensured as I did now.

But still whatever we do we can't ensure that defers will execute, because in case of panicking in a different goroutine, it crashes entire program.

Reply:
Ok got it. Let's merge this as is and we can iterate on printing stack traces.